apiary-metastore-listener/src/test/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListenerTest.java

```diff
@@ -68,36 +72,51 @@
   @Captor
   private ArgumentCaptor<PublishRequest> requestCaptor;
 
+  private Table table;
   private static final String TABLE_NAME = "some_table";
   private static final String DB_NAME = "some_db";
+  private static final String SOURCE_METASTORE_URIS = "thrift://remote_host:9883";
+  private static final List<String> PARTITION_VALUES = ImmutableList.of("value_1", "value_2", "value_3");
+  private static final List<String> PARTITION_COLUMNS = ImmutableList.of("Column_1", "Column_2", "Column_3");
+
+  private List<FieldSchema> partitionKeys;
   private ApiarySnsListener snsListener;
 
   @Before
   public void setup() {
     snsListener = new ApiarySnsListener(configuration, snsClient);
     when(snsClient.publish(any(PublishRequest.class))).thenReturn(publishResult);
+
+    FieldSchema partitionColumn1 = new FieldSchema("Column_1", "String", "");
+    FieldSchema partitionColumn2 = new FieldSchema("Column_2", "String", "");
+    FieldSchema partitionColumn3 = new FieldSchema("Column_3", "String", "");
+
+    partitionKeys = ImmutableList.of(partitionColumn1, partitionColumn2, partitionColumn3);
+
+    table = new Table();
+    table.setTableName(TABLE_NAME);
+    table.setDbName(DB_NAME);
+    table.putToParameters(METASTOREURIS.varname, SOURCE_METASTORE_URIS);
+    table.setPartitionKeys(partitionKeys);
+
   }
 
   @Test
   public void onCreateTable() throws MetaException {
     CreateTableEvent event = mock(CreateTableEvent.class);
     when(event.getStatus()).thenReturn(true);
-    Table table = new Table();
-    table.setTableName(TABLE_NAME);
-    table.setDbName(DB_NAME);
     when(event.getTable()).thenReturn(table);
 
     snsListener.onCreateTable(event);
     verify(snsClient).publish(requestCaptor.capture());
     PublishRequest publishRequest = requestCaptor.getValue();
-    assertThat(publishRequest.getMessage(),
-        is("{\"protocolVersion\":\""
-            + PROTOCOL_VERSION
-            + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\"}"));
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"sourceMetastoreUris\":\"thrift://remote_host:9883\"}"));
   }
 
   @Test
-  public void onInsert() throws MetaException {
+  public void onInsert() throws MetaException, NoSuchObjectException {
```

Comment:
Why is `NoSuchObjectException` now part of the signature?

Reply:
Removed. Left over from something I was trying to do with Insert event.